Branch-SDK/src/io/branch/referral/SystemObserver.java

```diff
@@ -164,7 +166,10 @@ private String getURIScheme(String packageName) {
                     xml = new byte[is.available()];
                     //noinspection ResultOfMethodCallIgnored
                     is.read(xml);
-                    scheme = new ApkParser().decompressXML(xml);
+                    JSONObject obj = new ApkParser().decompressXML(xml);
```

Comment:
SDK  call `getURIScheme()` method on install request. Adding the extended logic with XML parsing in this method does not look good. It will slow down the v1/install call unnecessarily. Can we move this APK parsing for intents into separate method and use it only for integration validation